hashtypechecker.cpp

```diff
@@ -0,0 +1,122 @@
+#include "hashtypechecker.h"
+
+#include <QtDebug>
+
+HashTypeChecker::HashTypeChecker()
+{
+    connect(&m_john,SIGNAL(finished(int, QProcess::ExitStatus)),this,SLOT(startParsing()));
+    connect (&m_john, SIGNAL(readyReadStandardOutput()), this, SLOT(processOutput())); // connect process signals with your code
```

Comment:
Remove the whitespace between connect and the open parentheses. It might also be helpful for you to enable the following settings in Qt Creator -> Preferences -> Text editor:
* "Clean whitespace"
* "Visualize whitespace"